ui/src/timeMachine/components/AggregationSelector.tsx

```diff
@@ -0,0 +1,94 @@
+// Libraries
+import React, {FunctionComponent} from 'react'
+import {connect, ConnectedProps} from 'react-redux'
+
+// Components
+import AggregationContents from 'src/timeMachine/components/AggregationContents'
+
+// Actions
+import {
+  selectBuilderFunction,
+  selectAggregateWindow,
+  setAggregateFillValues,
+  setIsAutoWindowPeriod,
+  setIsAutoFunction,
+} from 'src/timeMachine/actions/queryBuilder'
+
+// Utils
+import {
+  getActiveQuery,
+  getIsInCheckOverlay,
+  getActiveWindowPeriod,
+  getActiveTimeMachine,
+} from 'src/timeMachine/selectors'
+
+// Types
+import {AppState} from 'src/types'
+import {FUNCTIONS} from '../constants/queryBuilder'
+
+type ReduxProps = ConnectedProps<typeof connector>
+type Props = ReduxProps
+
+const AggregationSelector: FunctionComponent<Props> = ({
+  selectedFunctions,
+  isInCheckOverlay,
+  isAutoFunction,
+  isAutoWindowPeriod,
```

Comment:
all three of these props would be better served with a string rather than a boolean.  with booleans we are stuck with two states.  strings can represent any number of states i.e.

`isInCheckOverlay` => `location` `type SelectorLocation = 'check' | 'de' | 'flows'`
`isAutoFunction` => `funcMode` `type FuncSelectorMode = 'auto' | 'default' | 'userSelected'`
`isAutoWindowPeriod` => `winMode` `type WindowMode = 'auto' | 'none'`

these names / definitions are not meant to be exhaustive but illustrative.

Reply:
thanks watts!